tests/test_oml/test_functional/test_metrics/test_retrieval_metrics.py

```diff
@@ -275,3 +280,30 @@ def test_calc_fnmr_at_fmr_check_params(fmr_vals: Tuple[int, ...]) -> None:
         pos_dist = torch.zeros(10)
         neg_dist = torch.ones(10)
         calc_fnmr_at_fmr(pos_dist, neg_dist, fmr_vals)
+
+
+def test_metrics_with_categories() -> None:
+    retrieved_ids = LongTensor([[0, 5, 4], [2, 1, 5], [2, 1, 5]])
+    gt_ids = [LongTensor([0, 10, 4]), LongTensor([2, 3]), LongTensor([2, 3])]
+    query_categories = np.array(["cat", "dog", "dog"])
+
+    metrics = calc_retrieval_metrics(
+        retrieved_ids=retrieved_ids,
+        gt_ids=gt_ids,
+        query_categories=query_categories,
+        cmc_top_k=(1,),
+        precision_top_k=(3, 5),
+        map_top_k=tuple(),
+    )
+
+    assert math.isclose(metrics["cat"]["cmc"][1], 1)
```

Comment:
rework this assers: compare dicts instead (to check if there are unwilling extra keys)